src/Resolvers/Collections.ts

```diff
@@ -92,31 +92,42 @@ export class CollectionsResolver {
 
   @FieldResolver(type => [Collection])
   async relatedCollections(
+    @Arg("size", () => Int, { nullable: true }) size: number,
     @Root() collection: Collection
   ): Promise<Collection[]> {
+    const query: { where: {}; take?: number } = { where: {} }
     const isArtistBased =
       collection.query.artist_ids && collection.query.artist_ids.length > 0
 
+    if (size) {
+      query.take = size + 1
+    }
```

Comment:
I use plus one here because a little further down we use `reject` to remove the initial collection queried from the related collection. This ensures that when we pass 6 as the `size` parameter that we get 6 related collections back.

Reply:
I initially thought to have a default value for the size property in the event a parameter wasn't passed but I thought this might be a less intuitive API because as a user I would expect if no `size` arg is explicitly passed to get back all the related collections that are available.

Reply:
Is there a use-case for needing to return _every_ related collection? That will suffer from scaling problems unless there are other restrictions on the number of possible related collections.

I think a sensible default would be best here (we use `10` in gravity)! If there are existing use-cases that need more collections, we can also update those and follow-up here by adding the default in a subsequent PR.